### DISCUSSION TITLE
Possible to attach FileSink to "ADS-B Decoder"?

### DISCUSSION THREAD

**Discussion by DominikMair** [mhostetter/gr-adsb#42] [General]
Hi!
Thank you very much for this awesome work. I am new to gnuradio and tried to attach a file sink to your ADS-B Decoder with following Errors: 

_Error 0:
Connection (
  Block - adsb_decoder_0 - ADS-B Decoder(adsb_decoder)
    Source - decoded(decoded)
  Block - blocks_file_sink_0 - File Sink(blocks_file_sink)
    Sink - in(0)
):
  No connection known for domains "gr_message", "gr_stream"

Error 1:
Connection (
  Block - adsb_decoder_0 - ADS-B Decoder(adsb_decoder)
    Source - decoded(decoded)
  Block - blocks_file_sink_0 - File Sink(blocks_file_sink)
    Sink - in(0)
):
  Source IO size "0" does not match sink IO size "8"._

As i am not very familiar with gnuradio i am not sure if this is even possible with your Decoder block. Thus, I appreciate any help.

**Comment by mhostetter** [comment]
Hey. What is your goal with the file sink? For instance, what do you want/expect the file to contain?

You cannot connect a file sink (which only takes stream inputs) to the message output from the ADS-B Decoder. However, if you'd like the decoded stream of bits/bytes in the file, I believe you can connect a PDU to Tagged Stream block in between.

I can't test this right now, but I believe this will work.

![image](https://user-images.githubusercontent.com/12112573/120352154-80823300-c2ce-11eb-9694-44111b2e0442.png)
